core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraAdapter.kt

```diff
@@ -299,10 +390,17 @@ private fun buildZonePath(
         signalPositions.add(zonePathSignal.position)
     }
 
+    // Build chunk list
+    val chunks = MutableDirStaticIdxArrayList<TrackChunk>()
+    for (range in zoneTrackPath) {
+        val chunk = trackChunkMap[range.track]!![range.begin]!!
+        chunks.add(DirStaticIdx(chunk, range.direction.toKtDirection()))
+    }
+
     return builder.zonePath(
         entry, exit, zonePathLength,
         movableElements, movableElementsConfigs, movableElementsDistances,
-        signals, signalPositions,
+        signals, signalPositions, chunks
     )
```

Comment:
A side effects of having chunks is that the zone path becomes much lighter: we don't have to build a list of signals, signal positions, and switch positions; switch positions are transitions between track chunks, and signals can be stored in track chunks.

This can be done later, if you don't feel like making the change now (open an issue though)

Reply:
It requires a little work, some of this data is missing from chunks. This PR is already large enough, I'll open an issue